hnn_core/cell.py

```diff
@@ -341,6 +363,39 @@ def gid(self, gid):
         else:
             raise RuntimeError('Global ID for this cell already assigned!')
 
+    def distance_section(self, target_sec_name, base_sec_name, base_sec_end):
+        if (base_sec_name, base_sec_end) not in self.cell_tree.keys():
+            return 1000000  # Ask about this
+
+        # Children of the current section
+        curr_sec_children = self.cell_tree[(base_sec_name, base_sec_end)]
+        # All sections have 0 and 1 ends
+        end_pts = (0, 1)
+
+        # Base condition
+        # If target section is connected to current section
+        # Return (target section length / 2)
+        # As distances are measured till the centre of the target section
+        for end_pt in end_pts:
+            if (target_sec_name, end_pt) in curr_sec_children:
+                return self.sections[target_sec_name].L / 2
+
+        dist = 1000000  # Return large value
+
+        # Recursion to find distance
+        for sec_name, sec_end_pt in self.cell_tree[(base_sec_name,
+                                                    base_sec_end)]:
+            if (sec_name == base_sec_name):
+                dist_1 = (self.distance_section(target_sec_name, sec_name,
+                          sec_end_pt) + self.sections[sec_name].L)
+            else:
+                dist_1 = self.distance_section(target_sec_name,
+                                               sec_name, sec_end_pt)
```

Comment:
```suggestion
                dist_1 = self.distance_section(target_sec_name, node)
```

I don't fully follow the logic of these if/else statements

Reply:
The algorithm searches for the destination nodes (end 0 or 1 of the target section) in all available paths. If the target section is in the subtree of the current node dist_1 (now named it `dist_temp`) will contain distance between the current node to the centre of the target section